API/utils/Model.php

```diff
@@ -811,6 +811,121 @@ function RemoveAdmin($email) {
         $this->FreeStmt($stmt);
     }
 
+
+    /**
+     * @param int $projectId
+     * @param string $androidId
+     * @param string $term
+     * @param string $textBackTranslation
+     * @param string $audioRecordingFilename
+     * @param string $audioData
+     * @return int
+     */
+    public function CreateWordLinkRecording($projectId, $androidId, $term, $textBackTranslation, $audioRecordingFilename, $audioData) {
+        try {
+            $this->conn->beginTransaction();
+
+            $generatedFileName = $this->GenerateRecordingFileName($audioRecordingFilename);
+
+            // create record entry
+            $sql = "INSERT INTO WordLinkRecordings(term, projectId, textBackTranslation, fileName)
+				VALUES (?, ?, ?, ?);";
+            $stmt = $this->PrepareAndExecute($sql, array($term, $projectId, $textBackTranslation, $generatedFileName));
+            $this->FreeStmt($stmt);
+
+            $recordingId = $this->conn->lastInsertId();
+            $this->conn->commit();
+
+            # save recording file content
+            $directory = "Projects/$androidId/WordLinks";
+            PutFile($directory, $generatedFileName, $audioData);
+
+            return $recordingId;
+        } catch (\Exception $e) {
+            error_log($e->getMessage());
+            $this->conn->rollBack();
+            RespondWithError(500, $e->getMessage());
```

Comment:
Please do not leak the error message. Instead, return a generic error like "There was an exception creating the world link recording."